docs/integration/spring-cloud-gateway.md

```diff
@@ -0,0 +1,237 @@
+in advance.
+
+```java
+@RequestMapping("login")
+public Mono<String> login() {
+    return Mono.just("redirect:" + casdoorAuthService.getSigninUrl("http://localhost:9090/callback"));
+}
+```
+
+After Casdoor verification passed, it will be redirected to your application with code and state.
+
+You can get the code and call `getOAuthToken` method, then parse out jwt token.
+
+`CasdoorUser` contains the basic information about the user provided by Casdoor, you can use it as a keyword to set the
+session in your application.
+
+```java
+@RequestMapping("callback")
+public Mono<String> callback(String code,String state,ServerWebExchange exchange) {
+    String token = "";
+    CasdoorUser user = null;
+    try {
+        token = casdoorAuthService.getOAuthToken(code,state);
+        user = casdoorAuthService.parseJwtToken(token);
+    } catch(CasdoorAuthException e) {
+        e.printStackTrace();
+    }
+    CasdoorUser finalUser = user;
+    return exchange.getSession().flatMap(session -> {
+        session.getAttributes().put("casdoorUser", finalUser);
+        return Mono.just("redirect:/");
+    });
+}
+```
+
+Examples of APIs are shown below.
+
+- CasdoorAuthService
+  - `String token = casdoorAuthService.getOAuthToken(code, "app-built-in");`
+  - `CasdoorUser casdoorUser = casdoorAuthService.parseJwtToken(token);`
+- CasdoorUserService
+  - `CasdoorUser casdoorUser = casdoorUserService.getUser("admin");`
+  - `CasdoorUser casdoorUser = casdoorUserService.getUserByEmail("admin@example.com");`
+  - `CasdoorUser[] casdoorUsers = casdoorUserService.getUsers();`
+  - `CasdoorUser[] casdoorUsers = casdoorUserService.getSortedUsers("created_time", 5);`
+  - `int count = casdoorUserService.getUserCount("0");`
+  - `CasdoorResponse response = casdoorUserService.addUser(user);`
+  - `CasdoorResponse response = casdoorUserService.updateUser(user);`
+  - `CasdoorResponse response = casdoorUserService.deleteUser(user);`
+- CasdoorEmailService
+  - `CasdoorResponse response = casdoorEmailService.sendEmail(title, content, sender, receiver);`
+- CasdoorSmsService
+  - `CasdoorResponse response = casdoorSmsService.sendSms(randomCode(), receiver);`
+- CasdoorResourceService
+  - `CasdoorResponse response = casdoorResourceService.uploadResource(user, tag, parent, fullFilePath, file);`
+  - `CasdoorResponse response = casdoorResourceService.deleteResource(file.getName());`
+
+## Step7: Restart project
+
+After start, open your favorite browser and visit **http://localhost:9090**, then click any button which can request resources from `casdoor-api`.
+
+![index](/img/integration/spring_cloud_gateway/index.png)
+
+The gateway authentication logic will be triggered. Since you are not logged in, you will jump to the login interface. Click Login Button.
+
+![toLogin](/img/integration/spring_cloud_gateway/toLogin.png)
+
+You can see the unified login platform of Casdoor.
+
+![login](/img/integration/spring_cloud_gateway/login.png)
```

Comment:
<img width="546" alt="image" src="https://user-images.githubusercontent.com/53544726/185742794-fb956581-b7ff-4bee-a923-daa448b06151.png">

You can open Chrome Setting, change languages like this:
<img width="911" alt="image" src="https://user-images.githubusercontent.com/53544726/185742837-bd50c570-459a-4249-b5a4-c910ddcb160a.png">

Than the Casdoor login UI will display in English